narwhals/_arrow/group_by.py

```diff
@@ -36,7 +41,7 @@ def get_function_name_option(
     import pyarrow.compute as pc
 
     function_name_to_options = {
-        "count": pc.CountOptions(mode="all"),
+        "count": pc.CountOptions(mode="only_valid"),
```

Comment:
Hi Marco, I might be wrong, but, while trying to parametrize tests for agg methods in "simple" group_by(s), I've noticed that while this change is coherent with `polars`' `count` behaviour, it might not align with the behaviour of `polars`' `len`.

```
@pytest.mark.parametrize(
    ("agg_expr", "expected"),
    [
        (nw.col("b").len(), {"a": [1, 2], "b": [3, 1]}),
    ],
)
def test_group_by_aggr(
    constructor: Constructor,
    agg_expr: nw.Expr,
    expected: dict[str, list[int | float]],
) -> None:
    data = {"a": [1, 1, 1, 2], "b": [1, None, 2, 3]}
    result = (
        nw.from_native(constructor(data)).group_by("a").agg(agg_expr).sort("a")
    )
    assert_equal_data(result, expected)
```
fails for `pyarrow`.

I understand, though, that the issue is rather on `len` - for which we miss native support in groupby - rather than on `count`.

Just wanted to bring this to your attention, hope it's relevant!

Reply:
thanks @AlessandroMiola ! you're right - I was thinking that we only supported `agg(nw.len())` (for which we are consistent across backends) but indeed, we also support `agg(nw.col('a').len())`, for which pyarrow skips nulls but shouldn't

were you interested in making a PR to fix this?

Reply:
Sure, I can try :)